modules/programs/zsh.nix

```
@@ -555,10 +575,18 @@ in
         ${dirHashesStr}
 
         ${optionalString cfg.enableSyntaxHighlighting
-          # Load zsh-syntax-highlighting last, after all custom widgets have been created
+          # Load zsh-syntax-highlighting after all custom widgets have been created
           # https://github.com/zsh-users/zsh-syntax-highlighting#faq
           "source ${pkgs.zsh-syntax-highlighting}/share/zsh-syntax-highlighting/zsh-syntax-highlighting.zsh"
         }
+        ${optionalString cfg.historySubstringSearch.enable
```

The default value for `historySubstringSearch` is an empty attribute set so the `enable` does not always exist.
```suggestion
        ${optionalString (cfg.historySubstringSearch.enable or false)
```